src/lib/testing/expect-saga.ts

```diff
@@ -17,24 +17,33 @@ interface WithReducerStage<State, Payload> {
   withReducer(reducer: Reducer<State, any>, initialState?: DeepPartial<State>): WithMocksStage<State, Payload>;
 }
 
-interface WithMocksStage<State, Payload> extends AssertionStage<State, Payload> {
+interface WithMocksStage<State, Payload> extends WhenDispatchedStage<State, Payload> {
   /**
    * Supply some mocks which will be used instead of the real functions.
    *
    * @param mocks The array of mocks to be used by the bound function.
    */
-  withMocks(mocks: Mocks<State>): AssertionStage<State, Payload>;
+  andMocks(mocks: Mocks<State>): WhenDispatchedStage<State, Payload>;
 }
 
-interface AssertionStage<State, Payload> extends DispatchActionStage<State, Payload> {
+interface WhenDispatchedStage<State, Payload> {
+  /**
+   * Dispatch an action to the store to start the saga.
+   *
+   * @param action The action to start the saga.
+   */
+  whenDispatched(action: Action<Payload>): AssertionStage<State>;
```

Comment:
From a fluent-reading perspective I would prefer something like `whenTriggeredBy`, but since there is no reference to `dispatch` in there, the leap might be too big to grasp.

Reply:
@HenriBeck Any idea about this? Otherwise good to merge.

You can also go ahead with the merge now and we can change this 1 word when we come up with something more appropriate.

Reply:
Yeah, I think it's important here that we make it clear that action is dispatched. Maybe the others will come up with a slightly better name for this, but for now, I think it's okay.